homeassistant/components/synology_dsm/camera.py

```diff
@@ -176,16 +178,17 @@ async def stream_source(self) -> str | None:
         if not self.available:
             return None
 
-        return self.camera_data.live_view.rtsp  # type: ignore[no-any-return]
+        return self.camera_data.live_view.rtsp
 
     async def async_enable_motion_detection(self) -> None:
         """Enable motion detection in the camera."""
         _LOGGER.debug(
             "SynoDSMCamera.enable_motion_detection(%s)",
             self.camera_data.name,
         )
+        assert self._api.surveillance_station is not None
```

Comment:
All these assertions feel odd/wrong. Asserts are more of an edge case exception that should be avoided. However, this PR adds quite a lot of them, which usually indicates a typing or logical problem that needs to be solved.